pkg/component/controller/extensions_controller.go

```diff
@@ -462,6 +462,15 @@ func (ec *ExtensionsController) Start(ctx context.Context) error {
 	}
 
 	go func() {
+		// The ec.Reconcile will only reconcile if it's the, starting the controller-manager
+		// before becoming leader may result in getting Charts that won't be reconciled at all.
+		// If none of the node has became leader, which is likely to happen if the existing
+		// has a non-graceful shut down, all the reconciliations will be skipped in every node.
+		ec.L.Info("Waiting to become leader before starting controller-manager")
+		for !ec.leaderElector.IsLeader() {
```

Comment:
Maybe use the leader elector's acquire/lost callbacks instead of sleeping? We probably also want to stop the controller gracefully the lease gets lost.